### PR TITLE
Feat: Publish Flow

### PR DESCRIPTION
**Pull request by mattxyzeth**
## Purpose

This PR implements the new publish flow. I've implemented the basics but it still needs proper layout, accessibility, and styling.q

## Screenshots

Screen Shot 2022-10-19 at 3 58 54 PM
Screen Shot 2022-10-19 at 3 59 24 PM
Screen Shot 2022-10-19 at 3 59 31 PM
Screen Shot 2022-10-19 at 3 59 45 PM
Screen Shot 2022-10-19 at 3 59 53 PM

## Testing Instructions

1. Enter metadata
2. Accept transaction

## Checklist

- [x] Title follows the [naming convention](../README.md#pull-requests).
- [x] Add a description.
- [x] Add screenshots (optional).
- [x] Add testing instructions.
- [x] Add at least one reviewer.
- [ ] Apply related labels (optional).
- [ ] Leave comments on changes you would like to discuss.
- [x] Remove changes that are not related to this pull request.
- [x] Remove debugging code.
- [x] Test your changes locally and on staging (if applicable).
- [ ] Ensure test coverage value in config matches actual coverage (if applicable).
- [x] Ensure the build is passing.

### CODE REVIEW

**Comment by karlpawlowicz** [comment]
One thing I noticed was when I added an image, it pushed the publish button and status below the fold. We will need to come up with a better UX, so that those are always visible.

File: components/PublishPage/PublishHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import { useContext } from 'react'
+import Link from 'next/link'
+import { ChevronRightIcon } from '@heroicons/react/20/solid'
+
+import LoggedInButtonPopUp from '@/components/LoggedInButtonPopUp'
+import Logo from '@/components/Logo'
+
+import { PublishContext } from '@/context/PublishContext'
+
+const steps = ['Compile', 'Mint', 'Complete']
```
Comment (karlpawlowicz) [inline components/PublishPage/PublishHeader.tsx]: To me "Details" makes more sense imo.

```suggestion
const steps = ['Details', 'Mint', 'Complete']
```

**Review by karlpawlowicz — APPROVED** [review]

File: components/PublishPage/PublishForm.tsx [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+              }
+
+              return prevState
+            })
+          }
+        )
+
+        const [error, tx] = await resolver<TransactionResponse>(
+          contract.publish(cid, metadataURI, {
+            value: ethers.utils.parseEther('0.1'),
```
Comment (mattxyzeth) [inline components/PublishPage/PublishForm.tsx]: switch back to use `price` var

File: components/PublishPage/PublishPreview.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+import { Popover, Transition } from '@headlessui/react'
+
+// TODO: fix import path
+import { BlockContainer } from '@crcls/blocksui-sdk/dist/index.js'
+
+import { GlobalContext } from '@/context/GlobalContext'
+import { PublishContext } from '@/context/PublishContext'
+
+const PublishPreview = () => {
+  const globalCtx = useContext(GlobalContext)
```
Comment (mattxyzeth) [inline components/PublishPage/PublishPreview.tsx]: destructure apiHost

File: components/PublishPage/PublishHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                    className={stepIdx === step ? 'text-green-600' : undefined}
+                  >
+                    {stepName}
+                  </span>
+                  {stepIdx !== steps.length - 1 ? (
+                    <ChevronRightIcon
+                      className="ml-4 h-5 w-5 text-neutral-300"
+                      aria-hidden="true"
+                    />
+                  ) : null}
```
Comment (mattxyzeth) [inline components/PublishPage/PublishHeader.tsx]: not needed

**Pull request merged**